### PR TITLE
Feat/kv telegrafconfig

### LINKED ISSUE #11725: kv: port telegrafs

### PR DESCRIPTION
**Pull request by goller**
Closes #11725 

_Briefly describe your proposed changes:_
Port telegraf configurations to the key value store interface.
Updated various error messages in both URM and telegraf.

  - [x] Rebased/mergeable
  - [ ] Tests pass
  - [x] http/swagger.yml updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: bolt/telegraf.go [hunk 1]
```diff
@@ -23,15 +23,13 @@ func (c *Client) initializeTelegraf(ctx context.Context, tx *bolt.Tx) error {
...
 func (c *Client) FindTelegrafConfigByID(ctx context.Context, id platform.ID) (tc *platform.TelegrafConfig, err error) {
-	op := OpPrefix + platform.OpFindTelegrafConfigByID
 	err = c.db.View(func(tx *bolt.Tx) error {
-		var pErr *platform.Error
-		tc, pErr = c.findTelegrafConfigByID(ctx, tx, id)
-		if pErr != nil {
-			pErr.Op = op
-			err = pErr
+		var pe *platform.Error
+		tc, pe = c.findTelegrafConfigByID(ctx, tx, id)
```
Comment (desa) [inline bolt/telegraf.go]: why not
```go
tc, err := c.findTelegrafConfigByID(ctx, tx, id)
```

File: kv/telegraf.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+	for _, item := range m {
+		tc, err := s.findTelegrafConfigByID(ctx, tx, item.ResourceID)
+		if err == ErrTelegrafNotFound { // Stale user resource mappings are skipped
+			continue
+		}
+		if err != nil {
+			return nil, 0, InternalTelegrafServiceError(err)
+		}
+
+		if tc != nil {
```
Comment (desa) [inline kv/telegraf.go]: why would this ever be nil if `err == nil`?

File: kv/telegraf.go [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+	return tc, err
+}
+
+func (s *Service) findTelegrafConfigByID(ctx context.Context, tx Tx, id influxdb.ID) (*influxdb.TelegrafConfig, error) {
+	encID, err := id.Encode()
+	if err != nil {
+		return nil, ErrInvalidTelegrafID
+	}
+
+	bucket, err := tx.Bucket(telegrafBucket)
```
Comment (desa) [inline kv/telegraf.go]: should this be `s.telegrafBucket(tx)`?

File: kv/telegraf.go [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+	if !tc.OrganizationID.Valid() {
+		return ErrInvalidTelegrafOrgID
+	}
+
+	v, err := marshalTelegraf(tc)
+	if err != nil {
+		return err
+	}
+
+	bucket, err := tx.Bucket(telegrafBucket)
```
Comment (desa) [inline kv/telegraf.go]: *ditto* should this be `s.telegrafBucket(tx)`?

File: kv/urm.go [hunk 5]
```diff
@@ -30,14 +80,13 @@ func filterMappingsFn(filter influxdb.UserResourceMappingFilter) func(m *influxd
...
-			return err
-		}
-		ms = mappings
-		return nil
+	var (
+		err error
+		ms  []*influxdb.UserResourceMapping
+	)
+	err = s.kv.View(func(tx Tx) error {
+		ms, err = s.findUserResourceMappings(ctx, tx, filter)
```
Comment (desa) [inline kv/urm.go]: I go back and forth on this shadowing stuff. My initial gut reaction is that it's confusing, but I'm not sure what a better pattern is.

Comment (desa) [reply kv/urm.go]: Maybe passing in the thing that is being retrieved would make it better. Something like

```go
func (s *Service) FindUserResourceMappings(ctx context.Context, filter influxdb.UserResourceMappingFilter, opt ...influxdb.FindOptions)  {
    ms []*influxdb.UserResourceMapping

    err := s.kv.View(func(tx Tx) error {
        return s.findUserResourceMapping(ctx,tx,filter,&ms)
    })

    if err != nil {
        return nil, err
    }

    return ms, nil
}
```

**Comment by goller** [comment]
@desa I fixed your review comments in 24034ef

**Pull request merged**